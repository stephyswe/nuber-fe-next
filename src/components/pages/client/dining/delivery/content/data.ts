export const deliveryContentData = [
  {
    type: 'item',
    title: 'In a rush?',
    subtitle: 'Hereâ€™s the fastest delivery for you',
  },
  {
    type: 'item',
    title: 'Only on Uber Eats',
  },
  {
    type: 'section',
    title: 'Pick it up for free',
    subtitle: 'Skip the fees when you order pickup',
    btnText: 'See map',
    image:
      'https://d4p17acsd5wyj.cloudfront.net/eatsfeed/pickup-homefeed-carousel/pickupcarousel_desktopweb.svg',
  },
  {
    type: 'item',
    title: 'Popular near you',
  },
  {
    type: 'item',
    title: 'New on Uber Eats',
    subtitle: 'Be one of the first to support them',
  },
  {
    type: 'section',
    title: 'Free nationwide shipping',
    subtitle: 'Gourmet eats, shipped free',
    image:
      'https://d4p17acsd5wyj.cloudfront.net/eatsfeed/shipment/shipment_carousel_background_image_desktop_web.svg',
  },
];
