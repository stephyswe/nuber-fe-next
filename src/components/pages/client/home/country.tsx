import { Link, Typography } from '@/components';

import { HomeMap } from '@/ui/maps/home';

type CountryWithMapProps = {
  data: {
    title: string;
    link: { href: string; title: string };
    mapCenter: { lat: number; lng: number };
    cities: CountryProps[];
  };
};

type CountryProps = {
  city: string;
  link: string;
};

export function CountryWithMap({ data }: CountryWithMapProps) {
  const { title, link, mapCenter, cities } = data;
  return (
    <div>
      <div className='flex items-end justify-between'>
        <Typography as='h2' variant='4xl'>
          {title}
        </Typography>
        <Link variant='linkBase2' href={link.href}>
          {link.title}
        </Link>
      </div>

      <div className='pt-6'>
        <HomeMap data={cities} mapCenter={mapCenter} />
      </div>
      <div className='mt-6 mb-4 grid grid-cols-2 gap-6 md:mb-0 md:grid-cols-4'>
        {cities.map((country: CountryProps, index: number) => (
          <Link key={index} href={country.link}>
            {country.city}
          </Link>
        ))}
      </div>
    </div>
  );
}

type CountryListProps = {
  data: {
    title: string;
    countries: string[];
  };
};

export const CountryList = ({ data }: CountryListProps) => (
  <div>
    <Typography as='h2' variant='4xl'>
      {data.title}
    </Typography>

    <div className='mt-6 mb-4 grid grid-cols-2 gap-6 md:mb-0 md:grid-cols-4'>
      {data.countries.map((country: string, index: number) => (
        <a key={index}>{country}</a>
      ))}
    </div>
  </div>
);

export const CountryListNew = ({ data }: any) => (
  <div className='mt-8 grid grid-flow-col grid-rows-1 gap-[16px_24px]'>
    {data.map((item: any) => (
      <Link key={item.title} href={item.link}>
        {item.title}
      </Link>
    ))}
  </div>
);
