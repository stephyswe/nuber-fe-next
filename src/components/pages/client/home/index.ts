export { CountryList, CountryWithMap } from './country';
export { EmploymentList } from './employment';
