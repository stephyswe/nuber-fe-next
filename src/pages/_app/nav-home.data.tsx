import { SvgLogin } from '@/ui/icons';

export const navHomeData = {
  auth: {
    register: {
      title: 'Sign up',
      link: '/auth/register',
    },
    login: {
      title: 'Log in',
      svg: <SvgLogin />,
      link: '/auth/login',
    },
  },
};
