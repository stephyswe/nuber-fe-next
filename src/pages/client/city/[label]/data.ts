const cityInfo = [
  {
    title: 'Göteborg food delivery and takeout',
    subtitle:
      'With 80 restaurants in Göteborg on Uber Eats, including 12Till12Gott, Stigbergs Pizzeria, and DINÉ Burgers - Femman , you’ll have your pick of places from which to order food online. Get food, from Fast Food to Breakfast And Brunch, from some of the best restaurants in Göteborg delivered to your door. If you’d prefer to get your takeout yourself, simply browse Göteborg restaurants offering pickup.',
  },
  {
    title: 'Göteborg restaurants that deliver',
    subtitle:
      'Uber Eats helps you find food delivery and pickup options from a wide selection of places to eat in Göteborg. Enter an address to browse Göteborg restaurants and cafes offering food delivery. See Göteborg restaurants on Uber Eats that you’ve never tried? View their menus and star ratings to help decide if you’d like to try their food.',
  },
  {
    title: 'Best food in Göteborg',
    subtitle:
      'On a quest to taste the best food in Göteborg? Search for famous restaurants in Göteborg or for your personal favorite places to eat in Göteborg to see if they offer food delivery with Uber Eats. Sometimes the best food is just what you’re craving so if you know what you’d like to eat, browse Göteborg restaurants that deliver by cuisine or dish.',
  },
];

const breadcrumb = [
  {
    title: 'Sweden',
    link: '/',
  },
  {
    title: 'Västra Götaland',
    link: '/client/region/västra-götaland',
  },
  {
    title: 'Göteborg',
    link: '/client/city/göteborg-västra-götaland',
  },
];

export const mockDataCity = {
  title: 'Food Delivery in Göteborg',
  breadcrumb,
  cityInfo,
};
