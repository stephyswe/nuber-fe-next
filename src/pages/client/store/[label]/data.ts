export const dishNewData = [
  {
    __typename: 'Dish',
    id: 1,
    name: 'Pizza Speciale',
    price: 115,
    photo: '',
    description: 'Juicy!',
    type: 'Pizza',
    options: [],
  },
  {
    __typename: 'Dish',
    id: 2,
    name: 'Räksallad',
    price: 115,
    photo: '',
    description:
      'Räkor, vitost, ägg, citron, grekiska kalamataoliver. Isbergssallad, färska tomater, gurka, lök, vitost och bröd ingår.',
    type: 'Sallader',
    options: [],
  },
  {
    __typename: 'Dish',
    id: 3,
    name: 'KycklingSallad',
    price: 115,
    photo:
      'https://tb-static.uber.com/prod/image-proc/processed_images/fbf500543dbcc1a91bcb8de08a66396f/ffd640b0f9bc72838f2ebbee501a5d4b.jpeg',
    description:
      'Kyckling, vitost, färsk paprika, ananas. Isbergssallad, färska tomater, gurka, lök, vitost och bröd ingår.',
    type: 'Sallader',
    options: [
      {
        __typename: 'DishOption',
        name: 'Spice Level',
        extra: null,
        choices: [
          {
            __typename: 'DishChoice',
            name: 'Little Bit',
            extra: null,
          },
          {
            __typename: 'DishChoice',
            name: 'Strong',
            extra: null,
          },
        ],
      },
      {
        __typename: 'DishOption',
        name: 'Pickle',
        extra: 1,
        choices: null,
      },
      {
        __typename: 'DishOption',
        name: 'Size',
        extra: null,
        choices: [
          {
            __typename: 'DishChoice',
            name: 'L',
            extra: 2,
          },
          {
            __typename: 'DishChoice',
            name: 'XL',
            extra: 5,
          },
        ],
      },
    ],
  },
];

export const dishData = {
  Pizza: [
    {
      __typename: 'Dish',
      id: 1,
      name: 'Pizza Speciale',
      price: 115,
      photo: '',
      description: 'Juicy!',
      type: 'Pizza',
      options: [],
    },
  ],
  Sallader: [
    {
      __typename: 'Dish',
      id: 2,
      name: 'Räksallad',
      price: 115,
      photo: '',
      description:
        'Räkor, vitost, ägg, citron, grekiska kalamataoliver. Isbergssallad, färska tomater, gurka, lök, vitost och bröd ingår.',
      type: 'Sallader',
      options: [],
    },
    {
      __typename: 'Dish',
      id: 3,
      name: 'KycklingSallad',
      price: 115,
      photo:
        'https://tb-static.uber.com/prod/image-proc/processed_images/fbf500543dbcc1a91bcb8de08a66396f/ffd640b0f9bc72838f2ebbee501a5d4b.jpeg',
      description:
        'Kyckling, vitost, färsk paprika, ananas. Isbergssallad, färska tomater, gurka, lök, vitost och bröd ingår.',
      type: 'Sallader',
      options: [
        {
          __typename: 'DishOption',
          name: 'Spice Level',
          extra: null,
          choices: [
            {
              __typename: 'DishChoice',
              name: 'Little Bit',
              extra: null,
            },
            {
              __typename: 'DishChoice',
              name: 'Strong',
              extra: null,
            },
          ],
        },
        {
          __typename: 'DishOption',
          name: 'Pickle',
          extra: 1,
          choices: null,
        },
        {
          __typename: 'DishOption',
          name: 'Size',
          extra: null,
          choices: [
            {
              __typename: 'DishChoice',
              name: 'L',
              extra: 2,
            },
            {
              __typename: 'DishChoice',
              name: 'XL',
              extra: 5,
            },
          ],
        },
      ],
    },
  ],
};
