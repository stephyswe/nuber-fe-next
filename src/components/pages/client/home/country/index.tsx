import { Link, Typography } from '@/components';

import { isTest } from '@/constant/env';
import { useDelivery } from '@/contexts';
import { Headline, LoadingHome } from '@/ui';
import { HeadlineProps } from '@/ui/headline';
import { HomeMap } from '@/ui/maps/home';

export type CountryWithMapProps = {
  data: {
    title: string;
    link: HeadlineProps['link'];
    mapCenter: { lat: number; lng: number };
    cities: CountryProps[];
  };
};

type CountryProps = {
  city: string;
  link: string;
};

export const CountryWithMap = ({ data }: CountryWithMapProps) => {
  const { isComplete } = useDelivery();
  if (!isComplete && !isTest) return <LoadingHome h='180' num={3} />;
  const { title, link, mapCenter, cities } = data;
  return (
    <div data-testid='ui-country-with-map'>
      <Headline noArrow title={title} link={link} />
      <div className='m-[24px_0]'>
        <HomeMap data={cities} mapCenter={mapCenter} />
      </div>
      <CountryWithMapList data={cities} />
    </div>
  );
};

export const CountryWithMapList = ({
  data,
}: {
  data: CountryWithMapProps['data']['cities'];
}) => (
  <div
    data-testid='ui-country-with-map-list'
    className='mt-6 mb-4 grid grid-cols-2 gap-6 md:mb-0 md:grid-cols-4'
  >
    {data.map(({ link, city }: CountryProps, index: number) => (
      <Link key={index} href={link}>
        {city}
      </Link>
    ))}
  </div>
);

type CountryListProps = {
  data: {
    title: string;
    countries: string[];
  };
};

export const CountryList = ({ data }: CountryListProps) => {
  const { isComplete } = useDelivery();
  if (!isComplete && !isTest) return <LoadingHome h='180' num={3} />;
  return (
    <div data-testid='ui-country-list'>
      <Typography as='h2' variant='4xl'>
        {data.title}
      </Typography>

      <div className='mt-6 mb-4 grid grid-cols-2 gap-6 md:mb-0 md:grid-cols-4'>
        {data.countries.map((country: string, index: number) => (
          <a key={index}>{country}</a>
        ))}
      </div>
    </div>
  );
};

export const CountryListNew = ({ data }: any) => (
  <div className='mt-8 grid grid-flow-col grid-rows-1 gap-[16px_24px]'>
    {data.map((item: any) => (
      <Link key={item.title} href={item.link}>
        {item.title}
      </Link>
    ))}
  </div>
);
