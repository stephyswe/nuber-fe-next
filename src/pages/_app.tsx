import { ApolloProvider } from '@apollo/client';
import { NextPage } from 'next';
import { AppProps } from 'next/app';
import { ReactElement, ReactNode } from 'react';

import '@/styles/globals.scss';
import '@/styles/uber.css';
import '@/styles/uber-fonts.css';

import { RootLayout } from '@/ui/layout';

import client from '../lib/apollo';

export type NextPageWithLayout = NextPage & {
  getLayout?: (page: ReactElement) => ReactNode;
};

type AppPropsWithLayout = AppProps & {
  Component: NextPageWithLayout;
};

function MyApp({ Component, pageProps }: AppPropsWithLayout) {
  // Use the layout defined at the page level, if available
  const getLayout = Component.getLayout ?? ((page) => page);

  return getLayout(
    <ApolloProvider client={client}>
      <RootLayout>
        <Component {...pageProps} />
      </RootLayout>
    </ApolloProvider>
  );
}

export default MyApp;
