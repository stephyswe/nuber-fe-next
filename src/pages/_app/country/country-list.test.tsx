import { render, screen, takeSnapshot } from 'test-utils';

import { CountryWithMapList } from '@/pages/_app/country';
import { HomeData } from '@/pages/_app/home.data';

const defaultData: any[] = [];

// ** Snapshot ** //

takeSnapshot(<CountryWithMapList data={defaultData} />);
takeSnapshot(<CountryWithMapList data={HomeData.countryData.cities} />);

// ** Default ** //

test('container has elements', () => {
  render(<CountryWithMapList data={defaultData} />);
  const element = screen.getByTestId('ui-country-with-map-list');
  expect(element).toBeInTheDocument();
  expect(element).toHaveClass(
    'mt-6 mb-4 grid grid-cols-2 gap-6 md:mb-0 md:grid-cols-4'
  );
});

// ** Props - Item - Data ** //

describe('props', () => {
  test('data', () => {
    const cities = HomeData.countryData.cities;

    render(<CountryWithMapList data={cities} />);

    // length of links
    const links = screen.getAllByRole('link');
    expect(links).toHaveLength(cities.length);

    // text and link in list item
    expect(links[0]).toHaveTextContent(cities[0].city);
    expect(links[0]).toHaveAttribute('href', cities[0].link);
  });
});

//TODO: add more tests for this component
