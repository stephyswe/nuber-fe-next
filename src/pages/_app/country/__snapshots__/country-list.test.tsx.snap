// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mt-6 mb-4 grid grid-cols-2 gap-6 md:mb-0 md:grid-cols-4"
  data-testid="ui-country-with-map-list"
/>
`;

exports[`renders correctly 2`] = `
<div
  className="mt-6 mb-4 grid grid-cols-2 gap-6 md:mb-0 md:grid-cols-4"
  data-testid="ui-country-with-map-list"
>
  <a
    className=""
    data-testid="component-link"
    href="/client/city/g%C3%B6teborg-v%C3%A4stra-g%C3%B6taland"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    Gothenburg
  </a>
  <a
    className=""
    data-testid="component-link"
    href="/client/city/malm%C3%B6-sk%C3%A5ne"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    Malm√∂
  </a>
  <a
    className=""
    data-testid="component-link"
    href="/client/city/stockholm-stockholm"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    Stockholm
  </a>
</div>
`;
